src/txmempool.cpp

```diff
@@ -837,3 +845,111 @@ const CTxMemPool::setEntries & CTxMemPool::GetMemPoolChildren(txiter entry) cons
+        rootit--;
+        if (stage.count(rootit)) {
+            // If the transaction is already staged for deletion, we know its descendants are already processed, so skip it.
+            it++;
+            continue;
+        }
+
+        // Iterate breadth-first over all descendants of transaction 'rootit'...
+        std::deque<txiter> todo; // List of hashes that we still need to process (descendants of 'rootit').
+        setEntries now; // Set of tx entries that will need to be added to stage if 'rootit' is included.
+        CAmount nowfee = 0; // Sum of the fees in 'now'.
+        size_t nowsize = 0; // Sum of the tx sizes in 'now'.
+        size_t nowusage = 0; // Sum of the memory usages of transactions in 'now'.
+        todo.push_back(rootit); // Add 'rootit' to the todo list, to initiate processing its children.
+        bool good = true; // Whether including 'rootit' (and all its descendants) is a good idea.
+
+        while (!todo.empty()) {
+            const txiter& itnow = todo.front();
+            if (now.count(itnow))
+                continue;
+
+            const uint256& hashnow = itnow->GetTx().GetHash();
+            if (protect.count(hashnow)) {
+                // If this transaction is in the protected set, we're done with 'rootit'.
+                good = false;
+                break;
+            }
+            // Add 'hashnow' to the 'now' set, and update its statistics.
+            now.insert(itnow);
+            nowfee += itnow->GetFee();
+            nowsize += itnow->GetTxSize();
+            nowusage += itnow->DynamicMemoryUsage();
+            todo.pop_front();
+
+            // Find dependencies of 'hashnow' and them to todo.
+            std::map<COutPoint, CInPoint>::iterator iter = mapNextTx.lower_bound(COutPoint(hashnow, 0));
+            while (iter != mapNextTx.end() && iter->first.hash == hashnow) {
+                const uint256& nexthash = iter->second.ptx->GetHash();
+                txiter nextit = mapTx.find(nexthash);
+                assert(nextit != mapTx.end());
+                if (!stage.count(nextit) && !now.count(nextit))
+                    todo.push_back(nextit);
+                iter++;
+            }
+        }
+
+        // Check that we aren't removing a set who's total feerate is below that of the set we're removing
+        // ... and then add now to stage
+        if (good && (double)nowfee * toadd.GetTxSize() <= (double)toadd.GetFee() * nowsize) {
+            stage.insert(now.begin(), now.end());
+            expsize -= nowusage;
+            bestFeeRateRemoved = std::max(bestFeeRateRemoved, CFeeRate(nowfee, nowsize));
+        }
+        it++;
+    }
+
+    if (expsize <= sizelimit) {
+        BOOST_FOREACH(const txiter& it, stage)
+            removeUnchecked(it);
+
+        trackRemovedOrAddFailed(bestFeeRateRemoved);
+        return true;
+    } else {
+        trackRemovedOrAddFailed(CFeeRate(toadd.GetFee(), toadd.GetTxSize()));
```

Comment:
Hmm? No a very high fee tx will always evict transactions with lower feerate even if it ends up evicting a very large package to do so.